### PR TITLE
Remove nearest temperature interpolation

### PR DESCRIPTION
**Pull request by gridley**
Hey Paul,

So, to review, this PR's goal is to remove some code that I believe is extraneous and also potentially confusing for new users. For instance, if someone sets a fuel temperature to 700K, the error will likely be something along the lines of saying there's no data library at or near that value, so you ordinarily have to either raise temperature_tolerance (and basically get the wrong answer) or switch to interpolation mode.

The motivating factor for nearest mode is for testing purposes. When we have a temperature near a library temperature, we want to just use that without interpolation. In particular, for regression tests with only one temperature available, we need this behavior.

This PR changes the default behavior up. Now, interpolation mode is ALWAYS used. Now, the interesting addition is that we never need to use nearest interpolation. By default now, we use nearest interpolation only if there is one temperature in a nuclide data library. Otherwise, interpolation is used. Moreover, if the temperature to interpolate to rounds to the same temperature that a nuclide XS library is available at, only that library will be used, without using bounding temperature libraries. I think that by properly handling these edge cases, we can remove nearest interpolation mode and reduce branching in our XS lookup code slightly. I have already adopted this approach on GPU and hope to get it into the CPU code to ensure reproducibility, as this does change the RNG stream. Namely, in cases that previously used nearest interpolation, we do one more RNG sample in the XS lookup kernel than before.

Currently, I have only tested that the edge case handling as mentioned above works correctly by printing the data libraries we  read out. I am yet to test whether this gives correct interpolation behavior, and am open to ideas as to how we can prove this is giving the right answer.

### CODE REVIEW

File: src/mgxs.cpp [hunk 1]
```diff
@@ -102,59 +102,27 @@ Mgxs::metadata_from_hdf5(hid_t xs_id, const std::vector<double>& temperature,
...
-                          std::round(available_temps[j + 1])) == temps_to_read.end()) {
-              temps_to_read.push_back(std::round((int) available_temps[j + 1]));
-            }
-            break;
-          }
+      if ((available_temps[j] <= temperature[i]) &&
+          (temperature[i] < available_temps[j + 1])) {
+        if (std::find(temps_to_read.begin(), temps_to_read.end(),
+              std::round(available_temps[j])) == temps_to_read.end()) {
+          temps_to_read.push_back(std::round((int)available_temps[j]));
```
Comment (gridley) [inline src/mgxs.cpp:116]: BTW, I think this cast to int then round thing isn't right to do, but it was already like that

File: src/mgxs.cpp [hunk 2]
```diff
@@ -102,59 +102,27 @@ Mgxs::metadata_from_hdf5(hid_t xs_id, const std::vector<double>& temperature,
...
         }
 
+        if (std::find(temps_to_read.begin(), temps_to_read.end(),
+              std::round(available_temps[j + 1])) == temps_to_read.end()) {
+          temps_to_read.push_back(std::round((int)available_temps[j + 1]));
```
Comment (gridley) [inline src/mgxs.cpp:121]: same here

**Comment by gridley** [comment]
Briefly testing this, I found either no noticeable impact or slight speed increases for nontrivial and trivial cases, respectively the lattice problem in the lattice regression tests and the pin cell example. Both originally used nearest mode interpolation, of course. Currently, I am running a test to compute keff vs temperature on a pin cell to very high accuracy to ensure we still gets the interpolation correct, then will do the same on the new code.
